src/Ocelot/LoadBalancer/LoadBalancers/RoundRobin.cs

```diff
@@ -2,43 +2,108 @@
 using Ocelot.Responses;
 using Ocelot.Values;
 
-namespace Ocelot.LoadBalancer.LoadBalancers
+namespace Ocelot.LoadBalancer.LoadBalancers;
+
+public class RoundRobin : ILoadBalancer
 {
-    public class RoundRobin : ILoadBalancer
+    private readonly Func<Task<List<Service>>> _servicesDelegate;
+    private readonly string _serviceName;
+    private readonly List<Lease> _leasing;
+
+    public RoundRobin(Func<Task<List<Service>>> services, string serviceName)
     {
-        private readonly Func<Task<List<Service>>> _servicesDelegate;
-        private readonly object _lock = new();
+        _servicesDelegate = services;
+        _serviceName = serviceName;
+        _leasing = new();
+    }
+
+    private static readonly Dictionary<string, int> LastIndices = new();
+    protected static readonly object SyncRoot = new();
 
-        private int _last;
+    public event EventHandler<LeaseEventArgs> Leased;
+    protected virtual void OnLeased(LeaseEventArgs e) => Leased?.Invoke(this, e);
 
-        public RoundRobin(Func<Task<List<Service>>> services)
+    public virtual async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
```

Comment:
This method is too long, I would split it

Reply:
Fixed, the method's code was split into 3 sub-methods
See https://github.com/ThreeMammals/Ocelot/pull/2111/commits/c4c7ae0ff2a891d64b4f81b6a6d051cc15e71211#diff-cc104d57540132d29a89de66a0c0db287f8a85c2b7d5c20ed4d7958652db5847